test/server/lib/validation/index.test.js

```diff
@@ -19,71 +18,61 @@ describe.skip('server/lib/validation/applications', () => {
+  it('should validate a correctly formatted app  name', () => {
+    expect(validator.isValidAppName('app name')).to.equal(true);
+    expect(validator.isValidAppName('valid hyphenated-name')).to.equal(true);
+    expect(validator.isValidAppName('valid quoted\'name')).to.equal(true);
     expect(stubLogger).to.have.been.calledThrice;
   });
-
-  it('should validate and return an error if contact name is not correctly formatted', () => {
-    const errors = {};
-    errors.fullName = errorManifest.fullName.incorrect;
-    expect(validator.isValidContactName('-日女子أَبْجَدِيّ')).to.be.rejectedWith(errors);
+  it('should validate and return an error for blank app name', () => {
+    expect(validator.isValidAppName('')).to.equal(false);
     expect(stubLogger).to.have.been.calledOnce;
   });
 
-  it('should validate and return an error for blank contact name', () => {
-    const errors = {};
-    errors.fullName = errorManifest.fullName.blank;
-    expect(validator.isValidContactName('')).to.be.rejectedWith(errors);
-    expect(validator.isValidContactName(undefined)).to.be.rejectedWith(errors);
-    expect(validator.isValidContactName(null)).to.be.rejectedWith(errors);
+  it('should validate a correctly formatted app description', () => {
+    expect(validator.isValidDescription('Lorem ipsum dolor sit amet consectetur adipiscing elit.')).to.equal(true);
+    expect(validator.isValidDescription('Lorem-ipsum HYPHEN allowed.')).to.equal(true);
+    expect(validator.isValidDescription('Lorem ipsum\'s QUOTE allowed.')).to.equal(true);
     expect(stubLogger).to.have.been.calledThrice;
   });
+  it('should validate and return an error for an invalid description', () => {
+    expect(validator.isValidDescription('')).to.equal(false);
+    expect(stubLogger).to.have.been.calledOnce;
+  });
 
-  it('should validate a correctly formatted email', () => {
-    expect(validator.isValidEmail('matt@matt.com')).to.eventually.equal(true);
-    expect(validator.isValidEmail('matt-matt@matt.com')).to.eventually.equal(true);
+  it('should validate a correctly formatted Url', () => {
+    expect(validator.isValidUrl('https://example.com')).to.equal(true);
+    expect(validator.isValidUrl('example.com')).to.equal(true);
     expect(stubLogger).to.have.been.calledTwice;
   });
+  it('should validate and return an error for an invalid Url', () => {
+    expect(validator.isValidUrl('')).to.equal(false);
+    expect(validator.isValidUrl('/example')).to.equal(false);
+    expect(validator.isValidUrl(null)).to.equal(false);
+    expect(stubLogger).to.have.been.calledThrice;
+  });
 
-  it('should validate and return an error if email address is not correctly formatted', () => {
-    const errors = {};
-    errors.email = errorManifest.email.incorrect;
-    expect(validator.isValidEmail('matt.com')).to.be.rejectedWith(errors);
+  it.skip('should validate a correctly formatted domain', () => {
+    // regex not completed
+    expect(validator.isValidDomain('d')).to.equal(true);
```

Comment:
Is that actually a valid domain?

Reply:
at the moment yes but the regex has not been written and will be done in add key

Reply:
Ah understood.